js/main.js

```diff
@@ -189,3 +250,106 @@ function closeModal(event) {
+var $reviewForm = document.querySelector('#review-form');
+
+function popReviewForm(event) {
+  var $closestLi = null;
+  var $reviewFormImg = document.querySelector('.review-form-img');
+  if (event.target.matches('.review-button')) {
+    $closestLi = event.target.closest('li');
+    var $imdbID = $closestLi.getAttribute('data-entry-id');
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', 'http://www.omdbapi.com/?i=' + $imdbID + '&apikey=fd3f5e28');
+    xhr.responseType = 'json';
+    xhr.addEventListener('load', function () {
+      var movieObj = xhr.response;
+      $reviewForm.elements.movieTitleForm.value = movieObj.Title;
+      $reviewForm.elements.posterUrlForm.value = movieObj.Poster;
+      if ($reviewForm.elements.posterUrlForm.value === 'N/A') {
+        $reviewFormImg.setAttribute('src', 'images/noposter.png');
+      } else {
+        $reviewFormImg.setAttribute('src', $reviewForm.elements.posterUrlForm.value);
+      }
+    });
+    xhr.send();
+  }
+}
+
+var $posterUrlForm = document.querySelector('input[name="posterUrlForm"]');
+var $reviewFormImg = document.querySelector('.review-form-img');
+$posterUrlForm.addEventListener('input', updatePosterPreview);
+
+function updatePosterPreview(event) {
+  $reviewFormImg.setAttribute('src', $posterUrlForm.value);
+}
+
+var $starRating = document.querySelector('.star-rating');
+$starRating.addEventListener('click', setStarRating);
+
+function setStarRating(event) {
+  var $stars = document.querySelectorAll('.star-icon');
+  if (event.target.matches('#one-star')) {
+    for (let i = 0; i < 1; i++) {
+      $stars[i].className = 'fas fa-star star-icon';
+    }
+    for (let i = 1; i < $stars.length; i++) {
+      $stars[i].className = 'far fa-star star-icon';
+    }
+    $starRating.setAttribute('id', 'one-star-rating');
+  } else if (event.target.matches('#two-star')) {
+    for (let i = 0; i < 2; i++) {
+      $stars[i].className = 'fas fa-star star-icon';
+    }
+    for (let i = 2; i < $stars.length; i++) {
+      $stars[i].className = 'far fa-star star-icon';
+    }
+    $starRating.setAttribute('id', 'two-star-rating');
+  } else if (event.target.matches('#three-star')) {
+    for (let i = 0; i < 3; i++) {
+      $stars[i].className = 'fas fa-star star-icon';
+    }
+    for (let i = 3; i < $stars.length; i++) {
+      $stars[i].className = 'far fa-star star-icon';
+    }
+    $starRating.setAttribute('id', 'three-star-rating');
+  } else if (event.target.matches('#four-star')) {
+    for (let i = 0; i < 4; i++) {
+      $stars[i].className = 'fas fa-star star-icon';
+    }
+    for (let i = 4; i < $stars.length; i++) {
+      $stars[i].className = 'far fa-star star-icon';
+    }
+    $starRating.setAttribute('id', 'four-star-rating');
+  } else if (event.target.matches('#five-star')) {
+    for (let i = 0; i < $stars.length; i++) {
+      $stars[i].className = 'fas fa-star star-icon';
+    }
+    $starRating.setAttribute('id', 'five-star-rating');
+  }
```

Comment:
This is pretty manual, I would instead suggest creating a custom attribute to signify how many stars clicking that star represents, then when it is clicked, you can retrieve the value of that attribute, parse it into a number, and loop that amount of times to create that amount of solid stars. Then, you can loop `5- amountOfStars` times to create the empty stars.

Reply:
Fixed it! Thank you for the tip!!